### PR TITLE
Send raw events

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Allows sending raw payloads to the Discord gateway.

# Details
Currently there is no way for any consumer to send a raw, not validated payload to the Discord gateway without having to make manual changes. For extension authors specifically, this could be incredibly hindering when certain API endpoints or entities are `internal` only. One specific case is with connecting to a voice channel, which requires sending a [Voice State Update Payload through the gateway](https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus.VoiceNext/VoiceNextExtension.cs#L102-L114) . Currently I do not believe that the lib exposes the functionality that VNext requires, which is why I'm proposing this alternative quick and easy solution for both ease and future compatibility (such as when new gateway events are published before we're able to update) 

# Changes proposed
* Adds the `SendPayloadAsync<T>(GatewayOpCode opCode, T data)` and `SendPayloadAsync(GatewayOpCode opCode, object? data = null)` methods. Both are marked with `Obsolete` attributes to discourage use.
* Renames `WsSendAsync` to `SendRawPayloadAsync`

# Notes
Entirely untested but there is zero reason why this shouldn't work in theory.

### CODE REVIEW

File: DSharpPlus/Clients/DiscordClient.WebSocket.cs [hunk 1]
```diff
@@ -490,27 +490,55 @@ internal async Task SendResumeAsync()
...
         {
             var info = await this.GetGatewayInfoAsync().ConfigureAwait(false);
             this.GatewayInfo = info;
             this.GatewayUri = new Uri(info.Url);
         }
 
-        internal async Task WsSendAsync(string payload)
+        internal async Task SendRawPayloadAsync(string rawJson)
         {
-            this.Logger.LogTrace(LoggerEvents.GatewayWsTx, payload);
```
Comment (akiraveliara) [inline DSharpPlus/Clients/DiscordClient.WebSocket.cs:504]: this naming was completely fine (and arguably better, payload describes what you *want* to do)

**Comment by Naamloos** [comment]
Even though this is by design, I am in no way whatsoever opposed to this change. Apart from aki's comment (the old method name was indeed clearer imo) it LGTM.

**Review by Naamloos — APPROVED** [review]

**Comment by Naamloos** [comment]
disregard disregarding, aki's review was about the same thing.

**Comment by akiraveliara** [comment]
the old parameter name or the old method name?

**Comment by OoLunar** [comment]
Merging

**Pull request merged**